.github/workflows/build_commits.yaml

```diff
@@ -0,0 +1,42 @@
+name: build_commits
+
+
+on: [pull_request]
+
+jobs:
+  build_commits:
+    name: build commit code
+    runs-on: ubuntu-latest
+    steps:
+      - name: Install Go
+        uses: actions/setup-go@v2.1.3
+        with:
+          go-version: 1.16.3
+      - name: Cache LLVM and Clang
+        id: cache-llvm
+        uses: actions/cache@v2.1.4
+        with:
+          path: $HOME/.clang
+          key: llvm-10.0
+      - name: Install LLVM and Clang
+        uses: KyleMayes/install-llvm-action@v1.2.1
+        with:
+          version: "10.0"
+          directory: $HOME/.clang
+          cached: ${{ steps.cache-llvm.outputs.cache-hit }}    
+      - name: Install libelf
+        run: |
+          sudo apt-get install libelf-dev
+      - name: Checkout code
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - name: git fetch commit code
+        run: |
+          git fetch origin pull/$PR/head:rebase-head
```

Comment:
Just had the realization this is probably not what we want to do though? `pull/$PR/head` refers to the last commit in the PR. So if we run a rebase against that, it actually does nothing. Instead, what we'd want to rebase on is the parent of the first commit in the PR, right?

I'd suggest using the [PR list commits GitHub API](https://docs.github.com/en/rest/reference/pulls#list-commits-on-a-pull-request) to retrieve that. Something like this (untested):
```sh
PR_COMMITS_API_JSON=$(curl \
  -H "Accept: application/vnd.github.v3+json" \
  -H "Authorization: Bearer ${{ secrets.GITHUB_TOKEN }}" \
  ${{ github.event.pull_request.commits_url }})
PR_FIRST_SHA=$(echo "$PR_COMMITS_API_JSON" | jq -r ".[0].sha")
PR_PARENT_SHA=$(git rev-parse "${PR_FIRST_SHA}^")
```
And then we use that as `rebase` base reference.

Reply:
Thanks for pointing this out. You are right that `pull/$PR/head` will point to the last commit in the PR whereas we want the first commit of the PR. 
I will try the mentioned solution.